spec/integration/layout_periphery_boundingbox_spec.rb

```diff
@@ -0,0 +1,39 @@
+require 'spec_helper'
+
+describe 'when configuring the header and footer background colors' do
+
+  class PeripheryExtensionTest < ::Asciidoctor::Pdf::Converter
+
+    def build_pdf_options doc, theme
+      # assert the theme will call us for header and footer
+      theme.header_background_color = "009246"
+      theme.header_border_width = 0
+      theme.footer_background_color = "CE2B37"
+      theme.footer_border_width = 0
+      theme.header_height = 160
+      theme.footer_height = 160
+      theme.page_margin = [ 160, 48, 160, 48]
+
+      super doc, theme
+    end
+
+  end
+
+  it "should draw the configured color across the whole periphery region", :integration => true do |test|
+
+    PeripheryExtensionTest.register_for 'pdf'
+
+    Asciidoctor.convert "Hello world",
```

Comment:
Please use the `to_pdf` helper.

Reply:
This one turns out to be tricky.
the to_pdf helper is either denying writing outside the spec/fixture or then doesn't write and returns the analyze oder reader response.
Maybe split this into to_pdf_reader / to_analyzer / to_pdf_file?

Reply:
I can change the implementation of to_pdf to handle any additional requirements.

However, it should still be writing somewhere inside the spec/output directory. That keeps the tests self-contained.

Reply:
`spec` as jail root should suffice to read from fixtures and write to output.